ADR/0035-artifact-streams.md

```diff
@@ -0,0 +1,150 @@
+`Artifact`. An example of this constructor can resemble the `TestSubjectConstructor` as proposed in
+[Upstreaming integration-service].
+
+`Artifact`s can also store references to the process which created the object (i.e. PipelineRun). As soon as a
+PipelineRun is created, an `Artifact` should be created to enable visibility into the ongoing process if
+an `ArtifactStream` is used for tracking or visualization.
+
+### ArtifactStream
+
+All `Artifact`s have an owner reference to a single `ArtifactStream`. The `ArtifactStream` keeps track of the latest
+produced `Artifact` that has been created owned by that stream (i.e. similar to a floating tag reference).
+
+A validation webhook will prevent an `Artifact` from being created unless its configuration matches that of its owning
+`ArtifactStream`.
+
+### ArtifactCollection
+
+An `ArtifactCollection` is a set of `Artifact`s that are grouped together. An `ArtifactCollection` can include
+references to other `ArtifactCollection`s to enable intermediate logical separation and aggregated groupings.
+Additionally, there are no restrictions around how many `ArtifactCollection`s a given `Artifact` may belong to.
+
+### ArtifactCollectionStream
+
+All `ArtifactCollection`s have an owner reference to a single `ArtifactCollectionStream`. The `ArtifactCollectionStream`
+defines the set of `ArtifactStream`s and `ArtifactCollectionStream`s that are allowed in the owned `ArtifactCollection`s.
+The `ArtifactCollectionStream` also defines whether any individual update to a contained stream results in the automatic
+generation of a new owned `ArtifactCollection`. The `ArtifactCollectionStream` keeps track of the latest produced
+`ArtifactCollection` owned by that stream.
+
+A validation webhook will prevent an `ArtifactCollection` from being created unless all contained `Artifact`s and
+`ArtifactCollection`s match its owning `ArtifactCollectionStream`.
+
+## Consequences
+
+- If any controller does not want to interface with the `Artifact*` CRs directly (i.e. to achieve independence
+  from other controllers within AppStudio), they will need to create an adapter pattern between the relevant
+  APIs and their own internal data model.
+- We should now have native ability to describe built artifact and their various supply chain related attestations.
+  This ability holds whether the artifact is built within the kubernetes cluster or outside of it.
+- We lose the simplification of having a common [Component] and [Application] data model requirement between
+  all controllers as the relationships with `Artifact`s and related resources are loose couplings
+- We will have to produce multiple adapters for the various controllers in order to achieve data model independence
+- While each of the controllers that compose AppStudio should be able to exist on their own, they can all
+  "come together" around a common description of produced artifacts in order to create a unified build, test,
+  and delivery pipeline.
```

Comment:
What is the rationale of the adapters? Data model independence? It looks somewhat controversially to me. We want independent controllers but creating dependencies with adapters (are they also controllers?). Will independent components be usable on its own?